exercises/palindrome-products/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+#[derive(Debug, PartialEq)]
+pub enum Error{
+    Empty,
+    RangeFailure,
+}
+#[derive(Debug, PartialEq)]
+pub struct ReturnVals{
+    pub result: i32,
+    pub factors: Vec<(i32, i32)>,
```

Comment:
It is not obvious to me what value there is in returning the factors here. Best case, there exist certain implementations for which one can get the factorization more or less for free when generating the palindrome. Worst case, the student has to go back and factorize their solutions after generation, which is at best an orthogonal problem.

I see that the canonical data has them. There's no real explanation of why. I therefore propose that for the Rust track, we diverge from the canonical data by dropping the factors from the implementation and the tests.

Reply:
If we do this, we'd end up with a much nicer type for Palindrome:

```rust
pub type Palindrome = u64;
```